### PR TITLE
Add security group for consul default https api port 8501

### PR DESCRIPTION
**Pull request by Laakso**
PR addresses part of issue https://github.com/hashicorp/terraform-aws-consul/issues/92

Add security group for consul default https api port 8501.
https://www.consul.io/docs/install/ports

### CODE REVIEW

**Review by brikis98 — COMMENTED** [review]
Thanks for the PR!

Sanity check: I imagine Consul doesn't listen on HTTPS by default... You'd have to enable it. So should opening up the HTTPS port be something that is enabled via a flag too? Or should we add, say, `enable_http_port` (default: true) and `enable_https_port` (default: false) params, and only add the corresponding SG rules if those bools are set to true?

**Comment by Laakso** [comment]
> Sanity check: I imagine Consul doesn't listen on HTTPS by default... You'd have to enable it.

Correct.

> So should opening up the HTTPS port be something that is enabled via a flag too? Or should we add, say, enable_http_port (default: true) and enable_https_port (default: false) params, and only add the corresponding SG rules if those bools are set to true?

Sounds reasonable. I can make those changes.

**Comment by Laakso** [comment]
I figured it makes more sense to make HTTPS API as optional and leave HTTTP API port as is.

File: main.tf [hunk 1]
```diff
@@ -77,6 +77,9 @@ module "consul_servers" {
   vpc_id     = data.aws_vpc.default.id
   subnet_ids = data.aws_subnet_ids.default.ids
 
+  # To create security group for consul HTTPS API. HTTPS API (Optional) Is off by default.
+  enable_https_port = var.enable_https_port
```
Comment (brikis98) [inline main.tf:81]:
```suggestion
  # If set to true, this allows access to the consul HTTPS API
  enable_https_port = var.enable_https_port
```

File: modules/consul-cluster/variables.tf [hunk 2]
```diff
@@ -224,6 +224,12 @@ variable "http_api_port" {
   default     = 8500
 }
 
+variable "https_api_port" {
+  description = "The port used by clients to talk to the HTTPS API"
```
Comment (brikis98) [inline modules/consul-cluster/variables.tf:228]:
```suggestion
  description = "The port used by clients to talk to the HTTPS API. Only used if enable_https_port is set to true."
```

File: modules/consul-cluster/variables.tf [hunk 3]
```diff
@@ -254,6 +260,12 @@ variable "enable_iam_setup" {
   default     = true
 }
 
+variable "enable_https_port" {
+  description = "HTTPS API (Optional) Is off by default. If true, security group for consul HTTPS API is created."
```
Comment (brikis98) [inline modules/consul-cluster/variables.tf:264]:
```suggestion
  description = "If set to true, allow access to the Consul HTTPS port defined via the https_api_port variable."
```

File: modules/consul-security-group-rules/main.tf [hunk 4]
```diff
@@ -144,6 +155,17 @@ resource "aws_security_group_rule" "allow_http_api_inbound_from_security_group_i
   security_group_id = var.security_group_id
 }
 
+resource "aws_security_group_rule" "allow_https_api_inbound_from_security_group_ids" {
+  count                    = var.allowed_inbound_security_group_count
```
Comment (brikis98) [inline modules/consul-security-group-rules/main.tf:159]:
```suggestion
  count                    = var.enable_https_port ? var.allowed_inbound_security_group_count : 0
```

File: modules/consul-security-group-rules/main.tf [hunk 5]
```diff
@@ -218,6 +240,16 @@ resource "aws_security_group_rule" "allow_http_api_inbound_from_self" {
   security_group_id = var.security_group_id
 }
 
+resource "aws_security_group_rule" "allow_https_api_inbound_from_self" {
+  type      = "ingress"
```
Comment (brikis98) [inline modules/consul-security-group-rules/main.tf:244]:
```suggestion
resource "aws_security_group_rule" "allow_https_api_inbound_from_self" {
  count     = var.enable_https_port ? 1 : 0
  type      = "ingress"
```

File: modules/consul-security-group-rules/variables.tf [hunk 6]
```diff
@@ -61,9 +61,20 @@ variable "http_api_port" {
   default     = 8500
 }
 
+variable "https_api_port" {
+  description = "The port used by clients to talk to the HTTPS API"
```
Comment (brikis98) [inline modules/consul-security-group-rules/variables.tf:65]:
```suggestion
  description = "The port used by clients to talk to the HTTPS API. Only used if enable_https_port is set to true."
```

File: modules/consul-security-group-rules/variables.tf [hunk 7]
```diff
@@ -61,9 +61,20 @@ variable "http_api_port" {
...
+}
+
 variable "dns_port" {
   description = "The port used to resolve DNS queries."
   type        = number
   default     = 8600
 }
 
+variable "enable_https_port" {
+  description = "HTTPS API (Optional) Is off by default. If true, security group for consul HTTPS API is created."
```
Comment (brikis98) [inline modules/consul-security-group-rules/variables.tf:77]:
```suggestion
  description = "If set to true, allow access to the Consul HTTPS port defined via the https_api_port variable."
```

File: variables.tf [hunk 8]
```diff
@@ -60,3 +60,8 @@ variable "spot_price" {
   default     = null
 }
 
+variable "enable_https_port" {
+  description = "HTTPS API (Optional) Is off by default. If true, security group for consul HTTPS API is created."
```
Comment (brikis98) [inline variables.tf:64]:
```suggestion
  description = "If set to true, allow access to the Consul HTTPS port defined via the https_api_port variable."
```

**Comment by Laakso** [comment]
Added suggested changes

**Review by brikis98 — APPROVED** [review]
LGTM, thank you! I'll kick off tests now.

**Comment by brikis98** [comment]
https://github.com/hashicorp/terraform-aws-consul/releases/tag/v0.8.2

**Pull request merged**